### PR TITLE
Kill active programmer if still alive after closing last IDE window

### PR DESCRIPTION
**Pull request by facchinm**
Fixes https://github.com/arduino/Arduino/issues/7498

### CODE REVIEW

File: app/src/processing/app/Base.java [hunk 1]
```diff
@@ -925,45 +926,20 @@ public void actionPerformed(ActionEvent actionEvent) {
...
-        //ignore
-      }
-      rebuildRecentSketchesMenuItems();
 
-      // Save out the current prefs state
-      PreferencesData.save();
-
-      // Since this wasn't an actual Quit event, call System.exit()
-      System.exit(0);
+      handleQuit();
```
Comment (matthijskooijman) [inline app/src/processing/app/Base.java]: Isn't this something for a separate commit? Calling `handleQuit()` here seems unrelated to killing the programmer?

File: app/src/processing/app/Base.java [hunk 2]
```diff
@@ -925,45 +926,20 @@ public void actionPerformed(ActionEvent actionEvent) {
...
-//        if (editor == editors[i]) {
-//          for (int j = i; j < editorCount-1; j++) {
-//            editors[j] = editors[j+1];
-//          }
-//          editorCount--;
-//          // Set to null so that garbage collection occurs
-//          editors[editorCount] = null;
-//        }
-//      }
+
```
Comment (matthijskooijman) [inline app/src/processing/app/Base.java]: Same here, this might be nicer in a separate commit.

**Comment by facchinm** [comment]
Going for the separate commits :wink:

File: app/src/processing/app/Base.java (hunk 1 above)
Comment (facchinm) [reply app/src/processing/app/Base.java]: Moved to own commit

File: app/src/processing/app/Base.java (hunk 2 above)
Comment (facchinm) [reply app/src/processing/app/Base.java]: Moved to own commit
